src/main/python/camdkit/framework.py

```diff
@@ -7,22 +7,188 @@
+  radial: typing.Tuple[float]
+  tangential: typing.Optional[typing.Tuple[float]] = None
+
+@dataclasses.dataclass
+class PerspectiveShift:
+  "Shift in x and y of the centre of projection of the virtual camera"
+  Cx: float
+  Cy: float
+  
+@dataclasses.dataclass
+class CentreShift:
+  "Shift in x and y of the centre of distortion of the virtual camera"
+  cx: float
+  cy: float
+
+@dataclasses.dataclass
+class GlobalPosition:
+  "Global ENU and geodetic coordinates"
+  E: float
+  N: float
+  U: float
+  lat0: float
+  lon0: float
+  h0: float
+
+@dataclasses.dataclass
+class Timestamp:
+  """
+  A 48-bit integer representing seconds, and a 32-bit integer representing nanoseconds, and an
+  optional 32-bit integer representing attoseconds elapsed since 00:00 January 1st 1970 epoch.
+  Reference: https://datatracker.ietf.org/doc/html/rfc8877
+  """
+  seconds: int
+  nanoseconds: int
+  attoseconds: typing.Optional[int] = None
+
+class BaseEnum(Enum):
+  "Base class for enumerations"
+
+  def __str__(self):
+    return self.value
+
+
+class SynchronizationSourceEnum(BaseEnum):
+  GENLOCK = "genlock"
+  VIDEO_IN = "videoIn"
+  PTP = "ptp"
+  NTP = "ntp"
+
+@dataclasses.dataclass
+class SynchronizationOffsets:
+  """
+  Offsets in seconds between sync and sample. Critical for e.g. frame remapping, or when using
+  different data sources for position/rotation and lens encoding
+  """
+  translation: typing.Optional[float] = None		
+  rotation: typing.Optional[float] = None		
+  encoders: typing.Optional[float] = None
+
+  def validate(self):
+    return all([isinstance(self.translation, float), 
+                isinstance(self.rotation, float), isinstance(self.encoders, float)])
+  
+  @staticmethod
+  def to_json(value: typing.Any) -> typing.Any:
+    return dataclasses.asdict(value)
+
+class TimingModeEnum(BaseEnum):
+  INTERNAL = "internal"
+  EXTERNAL = "external"
+
+class TimecodeFormat(BaseEnum):
+  TC_24  = "24"
+  TC_24D = "24D"
+  TC_25  = "25"
+  TC_30  = "30"
+  TC_30D = "30D"
+
+  @classmethod
+  def to_int(cls, value):
+    if value == cls.TC_24 or value == cls.TC_24D: return 24
+    if value == cls.TC_25: return 25
+    if value == cls.TC_30 or value == cls.TC_30D: return 30
+    raise ValueError
+  
+  @classmethod
+  def to_float(cls, value):
+    if value == cls.TC_24: return 24.0
+    if value == cls.TC_24D: return 23.976
+    if value == cls.TC_25: return 25.0
+    if value == cls.TC_30: return 30
+    if value == cls.TC_30D: return 29.97
+    raise ValueError
+  
+  @staticmethod
+  def from_string(value):
+    return TimecodeFormat(value)
+
+@dataclasses.dataclass
+class Timecode:
+  "Timecode is a standard for labeling individual frames of data in media systems."
+  hours: int
+  minutes: int
+  seconds: int
+  frames: int
+  format: TimecodeFormat
+
+@dataclasses.dataclass
+class Synchronization:
+  "Data structure for synchronisation data"
+  frequency: float
```

Comment:
What are the units of frequency? Is this the nominal frequency of the time source? How is it used?